lldb/include/lldb/API/SBSaveCoreOptions.h

```diff
@@ -53,6 +53,30 @@ class LLDB_API SBSaveCoreOptions {
   /// \return The output file spec.
   SBFileSpec GetOutputFile() const;
 
+  /// Add a thread to save in the core file.
+  ///
+  /// \param thread_id The thread ID to save.
+  void AddThread(lldb::tid_t thread_id);
```

Comment:
Do we want to do things by `lldb::tid_t` or do we want to pass in a `lldb::SBThread`? The reason I mention this is there are two integer identifiers from a SBThread:
```
  lldb::tid_t lldb::SBThread::GetThreadID() const;
  uint32_t lldb::SBThread::GetIndexID() const;
```
This API could and probably should be:
```
void AddThread(lldb::SBThread thread);
```
Because in order to get a lldb::tid_t users would need to have a SBThread already anyway. Then the user can't mess up by specifying the wrong ID (index ID instead of `lldb::tid_t`)

Reply:
I think SBThread, because it solves my issue where I'm returning a thread at an index but need to support an error case